x-pack/plugins/fleet/server/services/epm/archive/validation/validate.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import fs from 'fs';
+import path from 'path';
+
+// Provides global.Go runtime
+import './wasm_exec';
+// @ts-expect-error
+const go = new Go();
+
+const wasmBuffer = fs.readFileSync(path.join(__dirname, 'validator.wasm'));
```

Comment:
__question:__ Just curious, I see this module is about 12mb, I assume it's not optimized yet (with more aggressive build optimization flags, `wasm-opt` and friends)?

Reply:
Yep we still need to explore making this smaller and distribute it via a proper npm module. This is currently being built with go-wasm, but I think it would make sense to look at tinygo.